### ISSUE TITLE
Parse Typescript annotations in template expressions

### ISSUE THREAD

**Issue by Trinovantes** [vuejs/vue-eslint-parser#128]
Vue's toolchain (vite/vue-loader) have recently added support for Typescript annotations in template expressions:

```html
<template>
  <q-input
    debounce="250"
    outlined
    clearable
    :model-value="item"
    @update:model-value="(val: string) => updateItem(idx, val)"
  />
<template>
```

This can compile and run just fine with vite/webpack but eslint is reporting an error in VSCode because its parser does not seem to support Typescript annotations in expressions yet

![Screenshot 2021-09-21 003201](https://user-images.githubusercontent.com/458691/134112259-66ea6515-a455-4552-a425-3dff34e659fc.png)

Is it possible to extend this parser (used by `eslint-plugin-vue`) to also parse Typescript annotations in expressions?

**Comment by ota-meshi** [comment]
You need to specify ` "@typescript-eslint/parser"` in `parserOptions.parser`.

https://github.com/vuejs/vue-eslint-parser#parseroptionsparser

**Comment by Trinovantes** [comment]
This seems to be an issue with eslint's VSCode extension. I did not notice until now that `eslint` already processes this just fine from the command line
